go/contacts/contacts.go

```diff
@@ -225,47 +121,105 @@ func ResolveContacts(mctx libkb.MetaContext, provider ContactsProvider, contacts
 	}
 	contactAssertionsSeen := make(map[contactAssertionPair]struct{})
 
-	for i, c := range contacts {
-		if _, found := contactsFound[i]; found {
-			continue
-		}
+	if len(emailSet)+len(phoneSet) == 0 {
+		// There is nothing to resolve.
+		return res, nil
+	}
 
-		// Add e.g. "(Work)" labels to display labels if there are multiple
-		// components in a contact.
-		var addLabel = len(c.Components) > 1
-		for _, component := range c.Components {
-			if _, foundErr := errorComponents[component.ValueString()]; foundErr {
-				// Do not return error components. If server said they are
-				// invalid, they can't be used for SBS either.
-				continue
-			}
+	phones := make([]keybase1.RawPhoneNumber, 0, len(phoneSet))
+	emails := make([]keybase1.EmailAddress, 0, len(emailSet))
+	for phone := range phoneSet {
+		phones = append(phones, phone)
+	}
+	for email := range emailSet {
+		emails = append(emails, email)
+	}
+	providerRes, err := provider.LookupAll(mctx, emails, phones, regionCode)
+	if err != nil {
+		return res, err
```

Comment:
we want to quick return error here?

Reply:
oh that's interesting. I wonder what we should do when someone tries to sync contacts while being offline